dataprep-pipelines/tests/integration.rs

```diff
@@ -1,44 +1,80 @@
-    fs::remove_dir_all("test").await.unwrap();
-    fs::create_dir("test").await.unwrap();
+    fs::remove_dir_all(&*TEST_DIR).await.unwrap();
+    fs::create_dir(&*TEST_DIR).await.unwrap();
     // create input directory
-    fs::create_dir("test/input").await.unwrap();
-    let input_dir = PathBuf::from("test/input");
+    fs::create_dir(&*INPUT_DIR).await.unwrap();
     // create output directory
-    fs::create_dir("test/output").await.unwrap();
-    let output_dir = PathBuf::from("test/output");
+    fs::create_dir(&*OUTPUT_DIR).await.unwrap();
     // create final output directory for unpacked
-    fs::create_dir("test/unpacked").await.unwrap();
-    let unpacked_dir = PathBuf::from("test/unpacked");
-    let manifest_file = PathBuf::from("test/manifest.json");
-    // create a file in the input directory
-    fs::write("test/input/test.txt", b"test").await.unwrap();
+    fs::create_dir(&*UNPACKED_DIR).await.unwrap();
+}
 
+async fn transform_and_check() {
     // run the function
     println!("doing pack pipeline!");
     dataprep_pipelines::pipeline::pack_pipeline::pack_pipeline(
-        input_dir.clone(),
-        output_dir.clone(),
-        manifest_file.clone(),
+        INPUT_DIR.clone(),
+        OUTPUT_DIR.clone(),
+        MANIFEST_FILE.clone(),
         1073741824, // 1GB
         true,
     )
     .await
     .unwrap();
     println!("doing unpack pipeline!");
     dataprep_pipelines::pipeline::unpack_pipeline::unpack_pipeline(
-        output_dir,
-        manifest_file,
-        unpacked_dir.clone(),
+        OUTPUT_DIR.to_path_buf(),
+        MANIFEST_FILE.to_path_buf(),
+        UNPACKED_DIR.clone(),
     )
     .await
     .unwrap();
     // checks if two directories are the same
-    assert_paths(input_dir, unpacked_dir).unwrap();
+    assert_paths(&*INPUT_DIR, &*UNPACKED_DIR).unwrap();
+}
+
+// TODO: make a function that puts random data into a file with a given length
+// so the argumetns are like (path, length) and it reutrns nothign
+
+async fn create_directory_structure(path: &str, length: usize) {
+    // creating a directory at the given path
+    fs::create_dir(path).await.unwrap();
+    // creates length number of directories and files within the directory at the given path
+    for i in 0..length {
+        // new directory path made
+        let dir_path = format!("{}/{}", path, i);
+        // create directory in the new directory path
+        fs::create_dir(dir_path.clone()).await.unwrap();
+        // new file path made
+        let file_path = format!("{}/file{}", dir_path, i);
+        // file is then created in the directory
+        let mut file = fs::File::create(file_path).await.unwrap();
+        // random data
+        let random_data = rand::thread_rng().gen::<[u8; 32]>();
+        file.write_all(&random_data);
+    }
+}
+#[tokio::test]
+async fn test_create_directory_structure() {
+    // assert and check the created directory structure
+    create_directory_structure("test_directory", 10);
```

Comment:
so yeah now you need to run `transform_and_check` after create_directory_structure to make sure my code is doing the thing

Reply:
also you need to call setup_structure before

and you need to ensure that create_directory is going into the proper `INPUT_DIR` subdirectory of `TEST_DIR`... does that make sense?

Reply:
Taken care of